include/vapor/Field.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * The base class of all possible fields for flow integration.
+ */
+
+#ifndef FIELD_H
+#define FIELD_H
+
+#include <glm/glm.hpp>
+#include <string>
+
+namespace flow
+{
+class Field
+{
+public:
+    // Constructor and destructor
+    Field();
+    virtual ~Field();
+
+    // 
+    // If a given position at a given time is inside of this field
+    //
+    virtual bool InsideVolumeVelocity( float time, const glm::vec3& pos ) = 0;
+    virtual bool InsideVolumeScalar(   float time, const glm::vec3& pos ) = 0;
+
+    //
+    // Retrieve the number of time steps in this field
+    //
+    virtual int GetNumberOfTimesteps() = 0;
```

Comment:
This is oen getter that could be made const, and now the change is already made. however, the others cannot be const simply because they're modifying the internal cache state.